tests/org.eclipse.papyrus.uml.interaction.model.tests/src/org/eclipse/papyrus/uml/interaction/model/tests/ModelEditFixture.java

```diff
@@ -0,0 +1,88 @@
+
+import java.util.Optional;
+
+import org.eclipse.emf.common.util.EList;
+import org.eclipse.gmf.runtime.notation.View;
+import org.eclipse.papyrus.uml.interaction.internal.model.impl.LogicalModelPlugin;
+import org.eclipse.papyrus.uml.interaction.model.MElement;
+import org.eclipse.papyrus.uml.interaction.model.MExecution;
+import org.eclipse.papyrus.uml.interaction.model.MInteraction;
+import org.eclipse.papyrus.uml.interaction.model.MLifeline;
+import org.eclipse.papyrus.uml.interaction.model.MMessage;
+import org.eclipse.papyrus.uml.interaction.model.MOccurrence;
+import org.eclipse.papyrus.uml.interaction.model.spi.DiagramHelper;
+import org.eclipse.papyrus.uml.interaction.model.spi.LayoutHelper;
+import org.eclipse.papyrus.uml.interaction.tests.rules.ModelFixture.Edit;
+import org.eclipse.uml2.uml.Element;
+import org.eclipse.uml2.uml.ExecutionSpecification;
+import org.eclipse.uml2.uml.InteractionFragment;
+import org.eclipse.uml2.uml.Message;
+
+@SuppressWarnings("restriction")
+public class ModelEditFixture extends Edit {
+
+	public DiagramHelper diagramHelper() {
+		return LogicalModelPlugin.getInstance().getDiagramHelper(this.getEditingDomain());
+	}
+
+	public LayoutHelper layoutHelper() {
+		return LogicalModelPlugin.getInstance().getLayoutHelper(this.getEditingDomain());
+	}
+
+	public int getLifelineBodyTop(MLifeline lifeline) {
+		View shape = lifeline.getDiagramView().orElse(null);
+		return layoutHelper().getTop(diagramHelper().getLifelineBodyShape(shape));
+	}
+
+	public MInteraction getMInteraction() {
+		return MInteraction.getInstance(getInteraction(), getSequenceDiagram().get());
+	}
+
+	public Optional<MMessage> getMMessage(Message message) {
+		return getMInteraction().getMessage(message);
+	}
+
+	public Optional<MExecution> getMExecution(ExecutionSpecification execution) {
+		Optional<MElement<? extends ExecutionSpecification>> element = getMInteraction()
+				.getElement(execution);
+		if (element.isPresent() && element.get() instanceof MExecution) {
+			return Optional.of((MExecution)element.get());
+		}
+		return Optional.empty();
+	}
+
+	public <T extends MElement<? extends Element>> T getElement(String qnFormat, String name, Class<T> type) {
+		String qName = String.format(qnFormat, name);
+
+		Element element = getElement(qName);
+		return Optional.ofNullable(element).flatMap(getMInteraction()::getElement).filter(type::isInstance)
+				.map(type::cast).orElseThrow(() -> new AssertionError("no such element: " + name)); //$NON-NLS-1$
+	}
+
+	@SuppressWarnings("unchecked")
+	public <T extends MElement<? extends Element>> T getElement(String qnFormat, String name) {
+		return (T)getElement(qnFormat, name, MElement.class);
+	}
+
+	@SuppressWarnings({"boxing" })
+	public Object isSemanticallyBefore(MOccurrence<? extends Element> one,
+			MOccurrence<? extends Element> other) {
+		EList<InteractionFragment> fragments = getMInteraction().getElement().getFragments();
```

Comment:
No, I don't think it would present a problem. I just purposefully bypassed the graph to be sure the assertion would test directly on the UML model, as it kind of is the goal of the test. However, since the vertex is essentially "only" a facade over the UML model, we might as well use it.